Modules/System/Base64 Convert/src/Base64ConvertImpl.Codeunit.al

```diff
@@ -57,19 +86,38 @@ codeunit 4111 "Base64 Convert Impl."
         exit(Base64String);
     end;
 
-    procedure FromBase64(Base64String: Text): Text
+    procedure FromBase64(Base64String: Text; TextEncoding: TextEncoding): Text
+    begin
+        exit(FromBase64(Base64String, TextEncoding, 1252));
+    end;
+
+    procedure FromBase64(Base64String: Text; TextEncoding: TextEncoding; CodePage: Integer): Text
     var
         Convert: DotNet Convert;
         Encoding: DotNet Encoding;
+        Windows: DotNet Encoding;
         OutputString: Text;
     begin
         if Base64String = '' then
             exit('');
 
-        OutputString := Encoding.UTF8().GetString(Convert.FromBase64String(Base64String));
+        case TextEncoding of
+            TextEncoding::UTF16:
+                OutputString := Encoding.Unicode().GetString(Convert.FromBase64String(Base64String));
+            TextEncoding::MSDos,
+            TextEncoding::Windows:
+                OutputString := Encoding.GetEncoding(CodePage).GetString(Convert.FromBase64String(Base64String));
```

Comment:
Why using the `CodePage` only if the encoding is Windows?

Shouldn't `FromBase64` take either `TextEncoding` or `CodePage`. I thought the code page defines the encoding uniquely.

Reply:
Since TextEncoding is the the "proper" way to specify encoding in BC, that should be the type, but TextEncoding have MsDos and Windows as types, are incomplete since they really need codepage to specify what "MsDos" and "Windows" actually mean.

Reply:
Almost there! Is this something we need to address before we go ahead and merge the PR?

Reply:
@hougaard, I admit I do not fully understand how encodings and code page work together.

Just to make sure I at least get something, would you explain what would be the result of the following:
`FromBase64('Some unicode text', TextEncoding::Windows, 1200)`

(1200 is the code page for UTF-16, according to [this doc](https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding))

And how does it differ from `FromBase64('Some unicode text', TextEncoding::UTF16)`?

Reply:
You got that part right.. The problem is, that your can do .CreateOutStream(strm,TextEncoding::Windows) and "Windows" (and MsDos) will be connected to whatever codepage your server is.

I'm trying to do 2 things:

1. Support the Textencoding Type so we're matching the overload from CreateInStream and CreateOutStream
2. Give an overload where you CAN specify what Windows or MsDos actually means.

An alternative would be to have an overload with just the codepage and let the overload with TextEncoding use "default". But in that case, I have no idea what default would be for Windows (properly 1252) and MsDos, but perhaps you can dig through the runtime source and find those?